osi_logicallane.proto

```diff
@@ -0,0 +1,471 @@
+    //   superElevation, then the t coordinate system is tilted along, so the T
+    //   coordinate is no longer calculated in the XY plane (as proposed for
+    //   OSI). It doesn't seem feasable to implement the same tilting for OSI,
+    //   so simulation tools will have to consider superElevation and convert
+    //   the T coordinate accordingly.
+    // - The angle will not be perfectly the same, due to the use of line
+    //   segments in OSI, and curves in OpenDRIVE. In the authors opinion, the
+    //   difference will be negligible if the poly_line is suitably sampled.
+    //
+    // Notes on design decisions:
+    // - The S coordinate is included directly, both for OpenDRIVE
+    //   compatibility, and to speed up calculations.
+    // - The rules on S coordinates (e.g. the calculation in 2D space) are
+    //   there to ensure OpenDRIVE compatibility.
+    // - The rules on T coordinates are there to ensure OpenDRIVE compatibility
+    //   for lanes without superelevation, and to make it easier to convert
+    //   between OSI and OpenDRIVE in case superelevation is present.
+    //
+    // TODO Describe needed precision.
+    // TODO Describe how large angles between line segments are allowed to be.
+    //
+    repeated ReferenceLinePoint poly_line = 2;
+
+    // A point on the reference line
+    message ReferenceLinePoint {
+        // A world position
+        optional Vector3d world_position = 1;
+
+        // S position on the reference line
+        optional double s_position = 2;
+    }
+}
+
+// Like a LaneBoundary, but with a reference and ST positions.
+//
+// Notes on design decisions:
+// - The LogicalLaneBoundary has ST coordinates, and is thus a separate type
+//   from LaneBoundary.
+//   Advantages of this decision:
+//   - Calculations like getting the lane width at a position are easy, since
+//     one can just get the boundary points of the left and right boundary at
+//     the desired S position (via linear interpolation), and calculate the
+//     width from the two points. Also getting something like the distance to
+//     the lane border is very easy.
+//   - No centerline of the lane is necessary, since this can very easily be
+//     generated from the boundaries.
+//   Disadvantages of this decision:
+//   - Lane boundaries cannot be shared with physical lanes. This results in
+//     more data needed. This can mostly be mitigated by only transmitting the
+//     lane boundaries during initialization (e.g. via the OSMP GroundTruthInit
+//     message).
+// - The LogicalLaneBoundary contains all data directly which an agent model is
+//   likely to need. It does not include information normally only used by
+//   sensor models (e.g. the exact length of the color markings on the road).
+//   This information can be gotten from the physical lane referenced in the
+//   LogicalLane, if needed.
+//
+// TODO To Discuss: It would be entirely possible to add ST coordinates also to
```

Comment:
Open to discuss if we adjust the  the initializing process, so that the user is able to decide which informaton are transmitted in online use?

Reply:
This should already work: _osi3::SensorViewConfiguration::omit_static_information_ can be set by the agent model, and should result in it not getting any logical lanes at runtime (but still getting the lane assignments of the objects).